homeassistant/components/wemo.py

```diff
@@ -0,0 +1,108 @@
+"""
+homeassistant.components.wemo
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+WeMo device discovery.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/wemo/
+"""
+import logging
+
+from homeassistant import bootstrap
+from homeassistant.components import discovery
+from homeassistant.const import (
+    EVENT_HOMEASSISTANT_STOP, EVENT_PLATFORM_DISCOVERED,
+    ATTR_SERVICE, ATTR_DISCOVERED)
+
+REQUIREMENTS = ['pywemo==0.3.12']
+
+DOMAIN = 'wemo'
+DISCOVER_LIGHTS = 'wemo.light'
+DISCOVER_MOTION = 'wemo.motion'
+DISCOVER_SWITCHES = 'wemo.switch'
+
+# mapping from Wemo model_name to service
+WEMO_MODEL_DISPATCH = {
+    'Bridge':  DISCOVER_LIGHTS,
+    'Insight': DISCOVER_SWITCHES,
+    'Maker':   DISCOVER_SWITCHES,
+    'Motion':  DISCOVER_MOTION,
+    'Switch':  DISCOVER_SWITCHES,
+}
+
+_WEMO_SUBSCRIPTION_REGISTRY = None
+
+_LOGGER = logging.getLogger(__name__)
+
+
+# pylint: disable=unused-argument, too-many-function-args
+def setup(hass, config):
+    """ Common set up for WeMo devices """
+    import pywemo
+
+    global _WEMO_SUBSCRIPTION_REGISTRY
+    if _WEMO_SUBSCRIPTION_REGISTRY is None:
+        _WEMO_SUBSCRIPTION_REGISTRY = pywemo.SubscriptionRegistry()
+        _WEMO_SUBSCRIPTION_REGISTRY.start()
+
```

Comment:
We had to do this trick in the previous code because the discovery function could be called multiple times.

I think you could just do
````
SUBSCRIPTION_REGISTRY = pywemo.SubscriptionRegistry()
````
And then import it from the other wemo modules.

We should change the name if we're going to use it in other modules.